// @coreui/angular/esm2022/lib/list-group/list-group-item.directive.mjs
import { booleanAttribute, computed, Directive, ElementRef, inject, input } from '@angular/core';
import * as i0 from "@angular/core";
export class ListGroupItemDirective {
    constructor() {
        this.hostElement = inject(ElementRef);
        /**
         * Toggle the active state for the component.
         * @type InputSignal<boolean | undefined>
         */
        this.active = input();
        /**
         * Sets the color context of the component to one of CoreUI’s themed colors.
         * @type InputSignal<boolean | undefined>
         */
        this.color = input();
        /**
         * Set disabled attr for the host element. [docs]
         * @type boolean
         */
        this.disabled = input(false, { transform: booleanAttribute });
        this.hostClasses = computed(() => {
            const host = this.hostElement.nativeElement;
            return {
                'list-group-item': true,
                'list-group-item-action': host.nodeName === 'A' || host.nodeName === 'BUTTON',
                active: !!this.active(),
                disabled: this._disabled(),
                [`list-group-item-${this.color()}`]: !!this.color()
            };
        });
        this._disabled = computed(() => this.disabled());
        this.ariaDisabled = computed(() => {
            return this._disabled() ? true : null;
        });
        this.attrDisabled = computed(() => {
            return this._disabled() ? '' : null;
        });
        this.tabIndex = computed(() => {
            return this._disabled() ? '-1' : null;
        });
        this.ariaCurrent = computed(() => {
            return this.active() || null;
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ListGroupItemDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "17.1.0", version: "18.2.2", type: ListGroupItemDirective, isStandalone: true, selector: "[cListGroupItem], c-list-group-item", inputs: { active: { classPropertyName: "active", publicName: "active", isSignal: true, isRequired: false, transformFunction: null }, color: { classPropertyName: "color", publicName: "color", isSignal: true, isRequired: false, transformFunction: null }, disabled: { classPropertyName: "disabled", publicName: "disabled", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "class": "hostClasses()", "attr.aria-disabled": "ariaDisabled()", "attr.aria-current": "ariaCurrent()", "attr.disabled": "attrDisabled()", "attr.tabindex": "tabIndex()" } }, exportAs: ["cListGroupItem"], ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ListGroupItemDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cListGroupItem], c-list-group-item',
                    exportAs: 'cListGroupItem',
                    standalone: true,
                    host: {
                        '[class]': 'hostClasses()',
                        '[attr.aria-disabled]': 'ariaDisabled()',
                        '[attr.aria-current]': 'ariaCurrent()',
                        '[attr.disabled]': 'attrDisabled()',
                        '[attr.tabindex]': 'tabIndex()'
                    }
                }]
        }] });
