// @coreui/angular/esm2022/lib/tabs-2/tabs.component.mjs
import { Component, effect, inject, input, model } from '@angular/core';
import { TabsService } from './tabs.service';
import * as i0 from "@angular/core";
let nextId = 0;
export class TabsComponent {
    constructor() {
        this.tabsService = inject(TabsService);
        /**
         * The active item key.
         * @type <string | number | undefined>
         */
        this.activeItemKey = model();
        /**
         * The id attribute
         * @type string
         */
        this.tabsId = `tabs-${nextId++}`;
        this.id = input(this.tabsId);
        this.activeItemEffect = effect(() => {
            this.tabsService.id.set(this.id());
            this.tabsService.activeItemKey.set(this.activeItemKey());
        }, { allowSignalWrites: true });
        this.tabsServiceEffect = effect(() => {
            this.activeItemKey.set(this.tabsService.activeItemKey());
        }, { allowSignalWrites: true });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: TabsComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.1.0", version: "18.2.2", type: TabsComponent, isStandalone: true, selector: "c-tabs", inputs: { activeItemKey: { classPropertyName: "activeItemKey", publicName: "activeItemKey", isSignal: true, isRequired: false, transformFunction: null }, id: { classPropertyName: "id", publicName: "id", isSignal: true, isRequired: false, transformFunction: null } }, outputs: { activeItemKey: "activeItemKeyChange" }, host: { properties: { "id": "id()" }, classAttribute: "tabs" }, providers: [TabsService], exportAs: ["cTabs"], ngImport: i0, template: '<ng-content />', isInline: true, styles: [":host{display:block}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: TabsComponent, decorators: [{
            type: Component,
            args: [{ exportAs: 'cTabs', selector: 'c-tabs', standalone: true, imports: [], template: '<ng-content />', providers: [TabsService], host: {
                        '[id]': 'id()',
                        class: 'tabs'
                    }, styles: [":host{display:block}\n"] }]
        }] });
