// @coreui/angular/esm2022/lib/image/img.directive.mjs
import { booleanAttribute, computed, Directive, input } from '@angular/core';
import * as i0 from "@angular/core";
export class ImgDirective {
    constructor() {
        /**
         * Set the horizontal aligment.
         * @type {'' | 'start' | 'end' | 'center'}
         */
        this.align = input('');
        /**
         * Make image responsive.
         * @type boolean
         */
        this.fluid = input(false, { transform: booleanAttribute });
        /**
         * Make image rounded.
         * @type boolean
         */
        this.rounded = input(false, { transform: booleanAttribute });
        /**
         * Give an image a rounded 1px border appearance.
         * @type boolean
         */
        this.thumbnail = input(false, { transform: booleanAttribute });
        /**
         * Color for image placeholder.
         */
        this.placeholderColor = input('transparent');
        this.hostStyles = computed(() => {
            return { backgroundColor: this.placeholderColor() };
        });
        this.hostClasses = computed(() => {
            const align = this.align();
            return {
                [`float-${align}`]: align === 'start' || align === 'end',
                'd-block': align === 'center',
                'mx-auto': align === 'center',
                'img-fluid': this.fluid(),
                rounded: this.rounded(),
                'img-thumbnail': this.thumbnail()
            };
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ImgDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "17.1.0", version: "18.2.2", type: ImgDirective, isStandalone: true, selector: "[cImg]", inputs: { align: { classPropertyName: "align", publicName: "align", isSignal: true, isRequired: false, transformFunction: null }, fluid: { classPropertyName: "fluid", publicName: "fluid", isSignal: true, isRequired: false, transformFunction: null }, rounded: { classPropertyName: "rounded", publicName: "rounded", isSignal: true, isRequired: false, transformFunction: null }, thumbnail: { classPropertyName: "thumbnail", publicName: "thumbnail", isSignal: true, isRequired: false, transformFunction: null }, placeholderColor: { classPropertyName: "placeholderColor", publicName: "placeholderColor", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "class": "hostClasses()", "style": "hostStyles()" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ImgDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cImg]',
                    standalone: true,
                    host: {
                        '[class]': 'hostClasses()',
                        '[style]': 'hostStyles()'
                    }
                }]
        }] });
