// @coreui/angular/esm2022/lib/utilities/shadow-on-scroll.directive.mjs
import { DOCUMENT } from '@angular/common';
import { DestroyRef, Directive, effect, ElementRef, inject, Input, signal } from '@angular/core';
import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
import { fromEvent } from 'rxjs';
import * as i0 from "@angular/core";
export class ShadowOnScrollDirective {
    #destroyRef = inject(DestroyRef);
    #document = inject(DOCUMENT);
    #elementRef = inject(ElementRef);
    #scrolled = signal(false);
    #scrollEffect = effect(() => {
        this.#elementRef.nativeElement.classList.toggle(this.#shadowClass, this.#scrolled());
    });
    #observable;
    #shadowClass = 'shadow-sm';
    constructor() {
        this.#destroyRef.onDestroy(() => {
            this.#scrollEffect?.destroy();
        });
    }
    set cShadowOnScroll(value) {
        this.#scrolled.set(false);
        if (value) {
            this.#shadowClass = value === true ? 'shadow' : `shadow-${value}`;
            this.#observable = fromEvent(this.#document, 'scroll')
                .pipe(takeUntilDestroyed(this.#destroyRef))
                .subscribe(scrolled => {
                this.#scrolled.set(this.#document.documentElement.scrollTop > 0);
            });
        }
        else {
            this.#observable?.unsubscribe();
        }
    }
    ;
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ShadowOnScrollDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "18.2.2", type: ShadowOnScrollDirective, isStandalone: true, selector: "[cShadowOnScroll]", inputs: { cShadowOnScroll: "cShadowOnScroll" }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ShadowOnScrollDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cShadowOnScroll]',
                    standalone: true
                }]
        }], ctorParameters: () => [], propDecorators: { cShadowOnScroll: [{
                type: Input
            }] } });
