// @coreui/angular/esm2022/lib/button-group/button-group/button-group.component.mjs
import { booleanAttribute, Component, computed, input } from '@angular/core';
import * as i0 from "@angular/core";
export class ButtonGroupComponent {
    constructor() {
        /**
         * Size the component small or large.
         * @type { 'sm' | 'lg' }
         */
        this.size = input();
        /**
         * Create a set of buttons that appear vertically stacked rather than horizontally. Split button dropdowns are not supported here.
         * @type boolean
         */
        this.vertical = input(false, { transform: booleanAttribute });
        /**
         * Default role attr for ButtonGroup. [docs]
         * @type InputSignal<string>
         * @default 'group'
         */
        this.role = input('group');
        this.hostClasses = computed(() => {
            return {
                'btn-group': !this.vertical(),
                'btn-group-vertical': this.vertical(),
                [`btn-group-${this.size()}`]: !!this.size()
            };
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ButtonGroupComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.1.0", version: "18.2.2", type: ButtonGroupComponent, isStandalone: true, selector: "c-button-group", inputs: { size: { classPropertyName: "size", publicName: "size", isSignal: true, isRequired: false, transformFunction: null }, vertical: { classPropertyName: "vertical", publicName: "vertical", isSignal: true, isRequired: false, transformFunction: null }, role: { classPropertyName: "role", publicName: "role", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "attr.role": "role()", "class": "hostClasses()" } }, ngImport: i0, template: '<ng-content />', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ButtonGroupComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'c-button-group',
                    template: '<ng-content />',
                    standalone: true,
                    host: { '[attr.role]': 'role()', '[class]': 'hostClasses()' }
                }]
        }] });
