// @coreui/angular/esm2022/lib/placeholder/placeholder-animation.directive.mjs
import { computed, contentChild, Directive, input } from '@angular/core';
import { PlaceholderDirective } from './placeholder.directive';
import * as i0 from "@angular/core";
export class PlaceholderAnimationDirective {
    constructor() {
        /**
         * Animation type for placeholder
         * @type 'glow' | 'wave'
         * @default undefined
         */
        this.animation = input(undefined, {
            alias: 'cPlaceholderAnimation'
        });
        this.placeholder = contentChild(PlaceholderDirective);
        this.hostClasses = computed(() => {
            return {
                [`placeholder-${this.animation()}`]: this.placeholder()?.visible() && !!this.animation()
            };
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: PlaceholderAnimationDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "17.2.0", version: "18.2.2", type: PlaceholderAnimationDirective, isStandalone: true, selector: "[cPlaceholderAnimation]", inputs: { animation: { classPropertyName: "animation", publicName: "cPlaceholderAnimation", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "class": "hostClasses()" } }, queries: [{ propertyName: "placeholder", first: true, predicate: PlaceholderDirective, descendants: true, isSignal: true }], ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: PlaceholderAnimationDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cPlaceholderAnimation]',
                    standalone: true,
                    host: {
                        '[class]': 'hostClasses()'
                    }
                }]
        }] });
