// @coreui/angular/esm2022/lib/progress/progress-bar.component.mjs
import { ChangeDetectionStrategy, Component, HostBinding, inject } from '@angular/core';
import { ProgressBarDirective } from './progress-bar.directive';
import * as i0 from "@angular/core";
import * as i1 from "./progress-bar.directive";
export class ProgressBarComponent {
    #progressBarDirective = inject(ProgressBarDirective, { optional: true });
    get hostClasses() {
        const animated = this.#progressBarDirective?.animated;
        const color = this.#progressBarDirective?.color;
        const variant = this.#progressBarDirective?.variant;
        return {
            'progress-bar': true,
            'progress-bar-animated': !!animated,
            [`progress-bar-${variant}`]: !!variant,
            [`bg-${color}`]: !!color
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ProgressBarComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "18.2.2", type: ProgressBarComponent, isStandalone: true, selector: "c-progress-bar", host: { properties: { "class": "this.hostClasses" }, classAttribute: "progress-bar" }, hostDirectives: [{ directive: i1.ProgressBarDirective, inputs: ["animated", "animated", "color", "color", "max", "max", "role", "role", "stacked", "stacked", "value", "value", "variant", "variant", "width", "width"] }], ngImport: i0, template: '<ng-content />', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ProgressBarComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'c-progress-bar',
                    template: '<ng-content />',
                    standalone: true,
                    hostDirectives: [
                        {
                            directive: ProgressBarDirective,
                            inputs: ['animated', 'color', 'max', 'role', 'stacked', 'value', 'variant', 'width']
                        }
                    ],
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    host: { class: 'progress-bar' }
                }]
        }], propDecorators: { hostClasses: [{
                type: HostBinding,
                args: ['class']
            }] } });
