// @coreui/angular/esm2022/lib/button/button.directive.mjs
import { booleanAttribute, computed, Directive, input } from '@angular/core';
import * as i0 from "@angular/core";
export class ButtonDirective {
    constructor() {
        /**
         * Toggle the active state for the component. [docs]
         * @type InputSignalWithTransform<boolean, unknown>
         */
        this.active = input(false, { transform: booleanAttribute });
        /**
         * Sets the color context of the component to one of CoreUI’s themed colors. [docs]
         * @type InputSignal<Colors>
         */
        this.color = input('primary');
        /**
         * Toggle the disabled state for the component.
         * @type InputSignalWithTransform<boolean, unknown>
         */
        this.disabled = input(false, { transform: booleanAttribute });
        /**
         * Select the shape of the component.
         * @type InputSignal<Shapes>
         */
        this.shape = input();
        /**
         * Size the component small or large.
         * @type InputSignal<'sm' | 'lg' | ''>
         */
        this.size = input('');
        /**
         * Specifies the type of button. Always specify the type attribute for the `<button>` element.
         * Different browsers may use different default types for the `<button>` element.
         * @type InputSignal<ButtonType>
         * @default 'button'
         */
        this.type = input('button');
        /**
         * Set the button variant to an outlined button or a ghost button.
         * @type InputSignal<'ghost' | 'outline' | undefined>
         */
        this.variant = input();
        this.hostClasses = computed(() => {
            return {
                btn: true,
                [`btn-${this.color()}`]: !!this.color() && !this.variant(),
                [`btn-${this.variant()}`]: !!this.variant() && !this.color(),
                [`btn-${this.variant()}-${this.color()}`]: !!this.variant() && !!this.color(),
                [`btn-${this.size()}`]: !!this.size(),
                [`${this.shape()}`]: !!this.shape(),
                active: this.active(),
                disabled: this._disabled()
            };
        });
        this._disabled = computed(() => this.disabled());
        this.ariaDisabled = computed(() => {
            return this._disabled() ? true : null;
        });
        this.attrDisabled = computed(() => {
            return this._disabled() ? '' : null;
        });
        this.tabIndex = computed(() => {
            return this._disabled() ? '-1' : null;
        });
        this.isActive = computed(() => {
            return this.active() || null;
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ButtonDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "17.1.0", version: "18.2.2", type: ButtonDirective, isStandalone: true, selector: "[cButton]", inputs: { active: { classPropertyName: "active", publicName: "active", isSignal: true, isRequired: false, transformFunction: null }, color: { classPropertyName: "color", publicName: "color", isSignal: true, isRequired: false, transformFunction: null }, disabled: { classPropertyName: "disabled", publicName: "disabled", isSignal: true, isRequired: false, transformFunction: null }, shape: { classPropertyName: "shape", publicName: "shape", isSignal: true, isRequired: false, transformFunction: null }, size: { classPropertyName: "size", publicName: "size", isSignal: true, isRequired: false, transformFunction: null }, type: { classPropertyName: "type", publicName: "type", isSignal: true, isRequired: false, transformFunction: null }, variant: { classPropertyName: "variant", publicName: "variant", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "class": "hostClasses()", "attr.aria-disabled": "ariaDisabled()", "attr.aria-pressed": "isActive()", "attr.disabled": "attrDisabled()", "attr.tabindex": "tabIndex()", "attr.type": "type()" }, classAttribute: "btn" }, exportAs: ["cButton"], ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ButtonDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cButton]',
                    exportAs: 'cButton',
                    standalone: true,
                    host: {
                        class: 'btn',
                        '[class]': 'hostClasses()',
                        '[attr.aria-disabled]': 'ariaDisabled()',
                        '[attr.aria-pressed]': 'isActive()',
                        '[attr.disabled]': 'attrDisabled()',
                        '[attr.tabindex]': 'tabIndex()',
                        '[attr.type]': 'type()'
                    }
                }]
        }] });
