// @coreui/angular/esm2022/lib/shared/theme.directive.mjs
import { booleanAttribute, Directive, ElementRef, inject, Input, Renderer2 } from '@angular/core';
import * as i0 from "@angular/core";
export class ThemeDirective {
    #hostElement = inject(ElementRef);
    #renderer = inject(Renderer2);
    /**
     * Add dark theme attribute.
     * @type 'dark' | 'light' | undefined
     */
    set colorScheme(scheme) {
        !!scheme ? this.setTheme(scheme) : this.unsetTheme();
    }
    ;
    /**
     * Add dark theme attribute.
     * @type boolean
     */
    set dark(darkTheme) {
        darkTheme ? this.setTheme('dark') : this.unsetTheme();
    }
    ;
    setTheme(theme) {
        if (theme) {
            this.#renderer.setAttribute(this.#hostElement.nativeElement, 'data-coreui-theme', theme);
        }
    }
    unsetTheme() {
        this.#renderer.removeAttribute(this.#hostElement.nativeElement, 'data-coreui-theme');
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ThemeDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "16.1.0", version: "18.2.2", type: ThemeDirective, isStandalone: true, selector: "[cTheme]", inputs: { colorScheme: "colorScheme", dark: ["dark", "dark", booleanAttribute] }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.2", ngImport: i0, type: ThemeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cTheme]',
                    standalone: true
                }]
        }], propDecorators: { colorScheme: [{
                type: Input
            }], dark: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });
